src/literal.js

```diff
@@ -14,22 +14,10 @@ class Literal extends Node {
       datatype = XSD.langString
     }
     // If not specified, a literal has the implied XSD.string default datatype
-    if (datatype) {
-      this.datatype = NamedNode.fromValue(datatype)
-    }
+    this.datatype = datatype ? NamedNode.fromValue(datatype) : XSD.string
   }
   copy () {
```

Comment:
This method seems rather nonsensical now, since indexing is done with the value, so either reindexing logic has to be created, or the user should pass intended modifications to this function.